MAINTAINERS.md

```diff
@@ -6,5 +22,91 @@
+
+## Zenhub Process Workflow
+We follow agile methodologies for our development and release process. We use zenhub to manage and track our stories and issues to effectively manage them over the sprint.
+
+### What is [Zenhub](https://www.zenhub.com/)?
+
+ZenHub is an agile project management and product roadmaps solution, natively integrated into GitHub. It is free to use for opensource repository and comes with a paid membership to manage private repositories. 
+We currently use zenhub only with our public and opensource repositories.
+
+### Setting up Zenhub
+
+Zenhub can be easily added as an [extension]((https://www.zenhub.com/extension)) to chrome and firefox which can be downloaded for free from the zenhub website. 
+Alternatively, we can use a the zenhub [webapp link](https://app.zenhub.com/workspaces/engineering-effectiveness-614cf4272a385f0015d2b48f/board?repos=357723952,406037663) to view the board.
+
+### Sprint Board aka Zenhub board aka Kanban board
+Once the zenhub extension is installed, zenhub board can be accessed using the Zenhub tab on github.
+
+#### Pipelines
+
+1. **New Issues -** Issues to be reviewed and estimated before being added to the Product Backlog.
+2. **Icebox -** Low priority Issues that do not need to be addressed in the near future**.**
+3. **Product Backlog -** Upcoming Issues that have been reviewed, estimated, and prioritized top-to-bottom.
+4. **Sprint Backlog -** Issues ready to be worked on in the sprint, prioritized top-to-bottom.
+5. **In Progress -** Issues currently being worked on by the team.
+6. **Review/QA -** Issues open to the team for review and testing. Code complete, pending feedback.
+7. **Done -** Issues tested and ready to be deployed to production. Verify the acceptance criteria and close the issue.
+8. **Closed -** Issues that are deployed to production and closed
+
+Description for each pipeline can also be found on the sprint board by clicking on the 3 dots next to the pipleine name.
+
+### Creating Issues
+
+Follow the steps below to create issues on Zenhub workflow -
+
+1. Create the issue for the desired repository following the required guidelines for mandatory fields on github.
+2. Add relevant tags to the issue. This would help us to track and filter issues.
+3. Select the correct pipeline for the issue (defaults to New Issues )
+4. Mark the issue for a sprint (if known)
+
+### Managing issues/stories
+
+1. Everyone working on a task should ensure the following -
+    1. Have an issue associated with it to record the work
+    2. The issue should be assigned
+    3. The issue should have an estimate set
+    4. The issue should have the correct pipeline
+2. Pull requests raised should be assigned and linked to their issues. There should not be a PR without an issue.
```

Comment:
`There should not be a PR without an issue.` - Could you provide more details why we have this practice? Inside of ZenHub pull requests are seen side-by-side with issues, so there doesn't seem to be a technical limitation.

If I spot a bug and fix it, I will be spending the time to write an issue and copy and paste the description into the PR which seems like needless duplication, is there a limitation that I'm overlooking?

Reply:
I believe spot fixing bugs which require **very very little** work or are only a few line changes maybe worked without an issue. 
However, for any minor changes which require some work, we should raise an issue. This is because it is easy to see and track who is working on what since the issue would be present on the board. However, a PR won't show up until it is raised and the team won't be informed.

Also, raising an issue would also help in tracking closed bugs for reporting in future which would not work with PRs.

Reply:
>Also, raising an issue would also help in tracking closed bugs for reporting in future which would not work with PRs.

I do not like to spend time on processes for the `"future"`, why don't we do this today?

Reply:
The document does not say we need to follow this in future. It needs to start today, as you mention for a better future.